XenAdmin/Wizards/NewNetworkWizard_Pages/NetWSriovDetails.cs

```diff
@@ -31,13 +62,23 @@ private void PopulateHostNicList(Host host, IXenConnection conn)
             comboBoxNicList.Items.Clear();
             foreach (PIF thePIF in conn.Cache.PIFs)
             {
-                if (thePIF.host.opaque_ref == host.opaque_ref && thePIF.IsPhysical() && !thePIF.IsBondNIC())
+                if (host != null && thePIF.host.opaque_ref == host.opaque_ref && thePIF.IsPhysical() && !thePIF.IsBondNIC() && thePIF.SriovCapable() && thePIF.sriov_physical_PIF_of.Count == 0)

```

Comment:
Now the comboBoxNicList can be empty (if there are no PIFs with `sriov_physical_PIF_of.Count == 0`) and the wizard allows users to click the Finish button, which will lead to a NullReferenceException.